contents/blog/improving-posthog-deployments.md

```diff
@@ -0,0 +1,99 @@
+
+* lint tests (via Helm lint): to verify if the Helm templates can be rendered without errors
+* unit tests (via [`quintush/helm-unittest`](https://github.com/quintush/helm-unittest)): to verify if the rendered Helm templates are behaving as we expect
+
+* integration tests:
+
+    * [`kubetest`](https://github.com/vapor-ware/kubetest): to verify if applying the rendered Helm templates against a Kubernetes target cluster gives us the stack we expect (example: are the disks encrypted? Can this pod communicate with this service?)
+
+    * [`k6`](https://k6.io/): HTTP test used to verify the reliability, performance and compliance of the PostHog installation (example: is the PostHog ingestion working correctly?)
+
+    * `e2e - k3s`: to verify Helm install/upgrade commands on a local k3s cluster
+
+    * `e2e - Amazon Web Services`, `e2e - DigitalOcean`, `e2e - Google Cloud Platform`: to verify Helm install command on the officially supported cloud platforms
+
+Thanks to those layers, we can now detect scenarios like:
+
+* template is invalid
+
+* template is valid, but it didn’t get rendered as we expect
+
+* template is valid and got rendered as we expect but doesn’t work on a specific Kubernetes version
+
+* template is valid and got rendered as we expect, works on all supported Kubernetes version, but it doesn’t work on a specific implementation of a cloud provider
+
+TODO: <insert cherry-picks of non-trivial CI failures>
+
+Those tests are helping us to identify and fix several bugs in our implementation, catching regressions before code gets released and enabling us to iterate faster than ever.
+
+For more information about the technical implementation, take a look [here](https://github.com/PostHog/charts-clickhouse#testing).
+
+
+## Goal 2: Improve built-in monitoring
+
+When you can successfully deploy a stack you are only half way in the journey. Making sure your stack is properly monitored is key in order to run and maintain a successful installation.
+
+To simplify this task for our self-hosting users, in the last couple of months we’ve improved the built-in monitoring we provide as part of PostHog by leveraging best in class open source components like Grafana, Prometheus and various service exporters.
+
+We create new templated dashboards, when we identify better metrics to monitor and techniques to debug an installation and make them available to everyone in the next release.
+
+Thanks to this work, you can now get critical  insights about the majority of PostHog services by simply enabling the monitoring features in the Helm chart (see our [docs](https://posthog.com/docs/self-host/deploy/configuration) for more info).
+
+
+## What’s next?
+
+Improving deployments for PostHog Cloud and our self-hosting customer is an effort we will keep investing in. Few of the projects we have in our roadmaps are:
+
+- keep improving our chart to align to Kubernetes best practices
+
+- add support for more cloud platforms
```

Comment:
Could we be more specific here? What platforms are we planning to support?

Reply:
There isn’t a specific plan (yet) but we now have a proper test infrastructure to support more of them without having to scale linearly our efforts.